code/datums/credit.dm

```diff
@@ -0,0 +1,17 @@
+/datum/credit
+	var/original_owner_name = "None"
+	var/pin = "0000"
+	var/acc_number = "12345"
+	var/balance = 0
+	var/special = 0
+
+/datum/credit/New(var/special = 0)
+	. = ..()
+	pin = "[rand(1000,9999)]"
+	acc_number = "[rand(10000, 99999)]"
```

Comment:
You might want to include some sort of safety catch for identical account numbers, since your `get_account(account_number)` proc simply returns the first matching datum with the supplied account number

I know the chances are low, but imagine two (or more!) accounts get assigned the same account number. The guy who gets added second would always be pulling the account of the first guy.

You'll want to check existing accounts for the account number before finalizing the account creation and adding it to the list of accounts. Maybe something like:
```var/unique = FALSE
while(!unique)
   acc_number = "[rand(10000, 99999)]"
   if(!get_account(acc_number))
      unique = TRUE
accounts.Add(src)```
So it will only add an account once it has verified the account number is not already associated with an existing account.

Reply:
i fixed the defines to make ERROR_NONEXISTANT_ACC equal 0